platform/ui/src/components/InputDoubleRange/InputDoubleRange.tsx

```diff
@@ -0,0 +1,223 @@
+  }, []);
+
+  const handleMouseUp = () => {
+    selectedThumbRef.current = null;
+  };
+
+  const handleGlobalMouseUp = () => {
+    // Remove global mouse event listeners
+    window.removeEventListener('mousemove', handleMouseMove);
+    window.removeEventListener('mouseup', handleGlobalMouseUp);
+    selectedThumbRef.current = null;
+  };
+
+  const handleMouseDown = e => {
+    const rect = sliderRef.current.getBoundingClientRect();
+    const x = e.clientX - rect.left;
+    const percentageClicked = (x / rect.width) * 100;
+
+    // Calculate the distances from the clicked point to both thumbs' positions
+    const distanceToStartThumb = Math.abs(percentageClicked - percentageStart);
+    const distanceToEndThumb = Math.abs(percentageClicked - percentageEnd);
+
+    // Check if the clicked point is within a threshold distance to either thumb
+    if (distanceToStartThumb < 10) {
+      selectedThumbRef.current = 0;
+    } else if (distanceToEndThumb < 10) {
+      selectedThumbRef.current = 1;
+    }
+
+    window.addEventListener('mousemove', handleMouseMove);
+    window.addEventListener('mouseup', handleGlobalMouseUp);
+  };
+
+  const handleMouseMove = e => {
+    const selectedThumbValue = selectedThumbRef.current;
+
+    if (selectedThumbValue === null) {
+      return;
+    }
+
+    const rect = sliderRef.current.getBoundingClientRect();
+    const x = e.clientX - rect.left;
+    const newValue =
+      Math.round(((x / rect.width) * (maxValue - minValue) + minValue) / step) * step;
+
+    // Make sure newValue is within [minValue, maxValue]
+    const clampedValue = Math.min(Math.max(newValue, minValue), maxValue);
+
+    // Ensure that left and right thumbs don't switch positions
+    if (selectedThumbValue === 0 && clampedValue >= rangeValue[1]) {
+      return;
+    }
+    if (selectedThumbValue === 1 && clampedValue <= rangeValue[0]) {
+      return;
+    }
+
+    // Update the correct values in the rangeValue array
+    const updatedRangeValue = [...rangeValue];
+    updatedRangeValue[selectedThumbValue] = clampedValue;
+    setRangeValue(updatedRangeValue);
+
+    onChange(updatedRangeValue);
+
+    // Update the thumb position
+    const percentage = Math.round(((clampedValue - minValue) / (maxValue - minValue)) * 100);
+    if (selectedThumbValue === 0) {
+      setPercentageStart(percentage);
+    } else {
+      setPercentageEnd(percentage);
+    }
+  };
+
+  // Calculate the range values percentages for gradient background
+  const rangeValuePercentageStart = ((rangeValue[0] - minValue) / (maxValue - minValue)) * 100;
+  const rangeValuePercentageEnd = ((rangeValue[1] - minValue) / (maxValue - minValue)) * 100;
+
+  return (
+    <div className={`flex select-none items-center space-x-2 ${containerClassName ?? ''}`}>
+      {showLabel && LabelOrEditableNumber(rangeValue[0], 0)}
+      <div
+        className="relative flex h-10 w-full items-center"
+        onMouseDown={handleMouseDown}
+        onMouseUp={handleMouseUp}
```

Comment:
It is great to see that we don't need the onMouseMove here any more. Do we still need the onMouseUp?